libr/asm/p/asm_mips_gnu.c

```diff
@@ -43,8 +45,37 @@ static int disassemble(struct r_asm_t *a, struct r_asm_op_t *op, const ut8 *buf,
 	Offset = a->pc;
 	memcpy (bytes, buf, 4); // TODO handle thumb
 
+	if ((a->cpu != pre_cpu) && (a->features != pre_features)) {
```

Comment:
uhm i think this is not necessary. because the cpu/precpu and features/prefeatures checks and sets are done below. so this is probably not necessary

Reply:
I think users may change asm.featur or asm.cpu when they try to see the code as different cpu（abi、mips cpu）

Reply:
without this block of code it is behaviing differently? arent those fields set below in case cpu or features has changed?

Reply:
1. Without this judgment, disasm_obj.disassembler_options will only be effective for the first time 【because it will only be set when a->features changes (in order to save some  @XVilka strstr())】
2. memset (&disasm_obj,'\0', sizeof (struct disassemble_info)) is necessary, the subsequent code may depend on NULL

Reply:
ok lets keep it for now :)